src/browserview/BrowserComponent.java

```diff
@@ -161,11 +161,16 @@ private void hidePageElements() {
 	 */
 	public void newLabel() {
 		logger.info("Navigating to New Label page");
-		runBrowserOperation(() -> {
-			if (!driver.getCurrentUrl().equals(GitHubURL.getPathForNewLabel())) {
-				driver.get(GitHubURL.getPathForNewLabel());
-			}
-		});
+		if (isBrowserActive()) {
```

Comment:
We could move this check into `runBrowserOperation`. That way we'd only have to update it in one place every time. Plus we're guaranteed that uses of `driver.getCurrentUrl()` won't be null/empty in the body of tasks, etc.